crates/sparrow-main/src/serve/compute_service.rs

```diff
@@ -101,21 +108,79 @@ impl ComputeService for ComputeServiceImpl {
-        Err(tonic::Status::internal("unsupported"))
+        let span = tracing::info_span!("StartMaterialization");
+        let _enter = span.enter();
+        let s3_helper = self.s3_helper.clone();
+        let id = request.get_ref().materialization_id.clone();
+
+        match start_materialization_impl(s3_helper, request.into_inner()) {
+            Ok(handle) => {
+                self.materializations.insert(id, handle);
+                Ok(Response::new(StartMaterializationResponse {}))
+            }
+            Err(e) => {
+                tracing::error!("failed to create materialization: {e}");
+                Err(tonic::Status::internal("failed to create materialization"))
+            }
+        }
     }
+
     async fn stop_materialization(
         &self,
-        _request: Request<StopMaterializationRequest>,
+        request: Request<StopMaterializationRequest>,
     ) -> Result<Response<StopMaterializationResponse>, Status> {
-        Err(tonic::Status::internal("unsupported"))
+        let span = tracing::info_span!("StopMaterialization");
+        let _enter = span.enter();
+        let id = request.into_inner().materialization_id;
+
+        if let Some(mut materialization) = self.materializations.get_mut(&id) {
+            match materialization.stop().await {
+                Ok(_) => Ok(Response::new(StopMaterializationResponse {})),
+                Err(e) => {
+                    tracing::error!("stop materialization error: {e}");
+                    Err(tonic::Status::internal(format!(
+                        "stop materialization error: {e}"
+                    )))
+                }
+            }
+        } else {
+            Err(tonic::Status::internal(format!(
+                "materialization {id} does not exist"
+            )))
+        }
     }
+
     async fn get_materialization_status(
         &self,
-        _request: Request<GetMaterializationStatusRequest>,
+        request: Request<GetMaterializationStatusRequest>,
     ) -> Result<Response<GetMaterializationStatusResponse>, Status> {
-        Err(tonic::Status::internal("unsupported"))
+        let span = tracing::info_span!("GetMaterializationStatus");
+        let _enter = span.enter();
+        let id = request.into_inner().materialization_id;
+
+        if let Some(materialization) = self.materializations.get(&id) {
+            let status = materialization.get_status();
+            let error = match status.error {
+                Some(e) => {
+                    tracing::error!("get materialization status error: {e}");
+                    e.to_string()
+                }
+                None => "".to_string(),
+            };
+
+            Ok(Response::new(GetMaterializationStatusResponse {
+                materialization_id: id,
+                state: status.state as i32,
+                progress: Some(status.progress),
+                error,
+            }))
+        } else {
+            Err(tonic::Status::internal(format!(
```

Comment:
If we want this to be clean, should probbaly be `tonic::Status::not_found` or some such... but probably OK for now.

Reply:
Good catch, updated